documentation/ui/transitions.md

```diff
@@ -1,21 +1,157 @@
+            // make sure you specify the size before using sharedElement modifier
+            .fillMaxSize()
+            .sharedElement(key = "sharedContainer")
+      ) { /** ... */ }
+   }
+}
+
+class NodeTwo(
+   buildContext: BuildContext
+) : Node(
+   buildContext = buildContext
+) {
+    
+   @Composable
+   override fun View(modifier: Modifier) {
+      Box(
+         modifier = Modifier
+             // make sure you specify the size before using sharedElement modifier
+            .requiredSize(64.dp)
+            .sharedElement(key = "sharedContainer")
+      ) { /** ... */ }
+   }
+}
+
+class ParentNode(
+   buildContext: BuildContext,
+   backStack: BackStack<NavTarget> = BackStack(
+      initialElement = NavTarget.Child1,
+      savedStateMap = buildContext.savedStateMap
+   )
+) : ParentNode<NavTarget>(
+   buildContext = buildContext,
+   navModel = backStack,
+) {
+    
+   override fun resolve(navTarget: NavTarget, buildContext: BuildContext): Node =
+      when (navTarget) {
+         NavTarget.Child1 -> NodeOne(buildContext)
+         NavTarget.Child2 -> NodeTwo(buildContext)
+      }
+   
+   @Composable
+   override fun View(modifier: Modifier) {
+      Children(
+         // or any other NavModel
+         navModel = backStack,
+         // or no transitionHandler at all. Using a slider will make the shared element slide away
+         // with the rest of of the content.
+         transitionHandler = rememberBackStackFader(),
+         withSharedElementTransition = true
+      )
+   }
+}
+
+```
+
+## Transitions with movable content
+
+You can move composable content between two Child Nodes without losing its state. You can only move
+content from a Node that is currently visible and transitioning to invisible state to a Node that
+is currently invisible and transitioning to visible state as movable content is intended to be
+composed once design and is moved from one part of the composition to another.
+
+To move content between two Child Nodes you need to use `localMovableContentWithTargetVisibility`
+composable function with the correct key to retrieve existing content if it exists or put content
+for this key if it doesn't exist. In addition to that, on Parent's `Children` composable you need to
+set `withMovableContent` to true.
 
+In the example below when a NodeOne is being replaced with NodeTwo in a BackStack or Spotlight NavModel
+`CustomMovableContent("movableContentKey")` will be moved from NodeOne to NodeTwo without losing its
+state.
+
+
+```kotlin
+@Composable
+fun CustomMovableContent(key: Any, modifier: Modifier = Modifier) {
+    localMovableContentWithTargetVisibility(key = key) {
+        // implement movable content here
+        var counter by remember(pageId) { mutableIntStateOf(0) }
+        LaunchedEffect(Unit) {
+            while (true) {
+                delay(1000)
+                counter++
+            }
+        }
+        Text(text = "$counter")
+    }?.invoke()
+}
+
+// NodeOne 
```

Comment:
This isn't copy-pasteable, do you mind adding all the code so that it's plugable and people can play with it?